MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/AdsImaSDKListener.java

```diff
@@ -0,0 +1,75 @@
+package com.mux.stats.sdk.muxstats;
+
+import com.google.ads.interactivemedia.v3.api.AdErrorEvent;
+import com.google.ads.interactivemedia.v3.api.AdEvent;
+import com.mux.stats.sdk.core.events.playback.AdBreakEndEvent;
+import com.mux.stats.sdk.core.events.playback.AdBreakStartEvent;
+import com.mux.stats.sdk.core.events.playback.AdEndedEvent;
+import com.mux.stats.sdk.core.events.playback.AdFirstQuartileEvent;
+import com.mux.stats.sdk.core.events.playback.AdMidpointEvent;
+import com.mux.stats.sdk.core.events.playback.AdPauseEvent;
+import com.mux.stats.sdk.core.events.playback.AdPlayEvent;
+import com.mux.stats.sdk.core.events.playback.AdPlayingEvent;
+import com.mux.stats.sdk.core.events.playback.AdRequestEvent;
+import com.mux.stats.sdk.core.events.playback.AdResponseEvent;
+import com.mux.stats.sdk.core.events.playback.AdThirdQuartileEvent;
+
+public class AdsImaSDKListener implements AdErrorEvent.AdErrorListener, AdEvent.AdEventListener {
+    private MuxBaseExoPlayer exoPlayerListener;
+
+    public AdsImaSDKListener(MuxBaseExoPlayer listener) {
+        exoPlayerListener = listener;
+    }
+
+    @Override
+    public void onAdError(AdErrorEvent adErrorEvent) {
+        if (exoPlayerListener != null)
+            exoPlayerListener.dispatch(new com.mux.stats.sdk.core.events.playback.AdErrorEvent(null));
+    }
+
+    @Override
+    public void onAdEvent(AdEvent adEvent) {
+        if (exoPlayerListener != null) {
+            switch (adEvent.getType()) {
+                case AD_BREAK_STARTED:
+                    exoPlayerListener.dispatch(new AdBreakStartEvent(null));
+                    break;
+                case AD_BREAK_ENDED:
+                    exoPlayerListener.dispatch(new AdBreakEndEvent(null));
+                    break;
+                case STARTED:
+                    exoPlayerListener.dispatch(new AdPlayEvent(null));
+                    exoPlayerListener.dispatch(new AdPlayingEvent(null));
+                    break;
+                case FIRST_QUARTILE:
+                    exoPlayerListener.dispatch(new AdFirstQuartileEvent(null));
+                    break;
+                case MIDPOINT:
+                    exoPlayerListener.dispatch(new AdMidpointEvent(null));
+                    break;
+                case THIRD_QUARTILE:
+                    exoPlayerListener.dispatch(new AdThirdQuartileEvent(null));
+                    break;
+                case COMPLETED:
+                    exoPlayerListener.dispatch(new AdEndedEvent(null));
+                    break;
+                case PAUSED:
+                    exoPlayerListener.dispatch(new AdPauseEvent(null));
+                    break;
+                case RESUMED:
+                    exoPlayerListener.dispatch(new AdPlayingEvent(null));
+                    break;
+                default:
+                    break;
+            }
+        }
+    }
+
+    public void onAdRequested() {
```

Comment:
The IMA SDK doesn't expose these ad events to us, does it? Is there a call the client has to take in their application that they can then at that point tell us?

Reply:
I'd also call this `onAdRequest` and the one below to just `onAdResponse`

Reply:
I think they will have to signal to us when the request is made (which is fine), but the response I think we can capture ourselves.

Reply:
fixed.